core/src/main/java/tc/oc/pgm/scoreboard/SidebarMatchModule.java

```diff
@@ -466,48 +473,59 @@ private void renderSidebar() {
       }
 
       for (Competitor competitor : sortedCompetitors) {
-        if (!firstTeam) {
+        if (!(firstTeam || isSuperCompactWool)) {
           // Add a blank row between teams
           rows.add("");
         }
         firstTeam = false;
 
-        // Add a row for the team name
-        rows.add(competitor.getStyledName(NameStyle.FANCY).render().toLegacyText());
-
-        if (isCompactWool()) {
-          String woolText = " ";
-          boolean firstWool = true;
-
-          List<Goal> sortedWools = new ArrayList<>(gmm.getGoals(competitor));
-          Collections.sort(
-              sortedWools,
-              new Comparator<Goal>() {
-                @Override
-                public int compare(Goal a, Goal b) {
-                  return a.getName().compareToIgnoreCase(b.getName());
+        // Prevent team name from showing up if there is no space for its objectives
+        if (!isSuperCompactWool || rows.size() + 2 < MAX_ROWS) {
+
+          // Add a row for the team name
+          rows.add(competitor.getStyledName(NameStyle.FANCY).render().toLegacyText());
+
+          if (isCompactWool) {
+            String woolText = " ";
+            boolean firstWool = true;
+
+            List<Goal> sortedWools = new ArrayList<>(gmm.getGoals(competitor));
+            Collections.sort(
+                sortedWools,
+                new Comparator<Goal>() {
+                  @Override
+                  public int compare(Goal a, Goal b) {
+                    return a.getName().compareToIgnoreCase(b.getName());
+                  }
+                });
```

Comment:
You should use lambda here. Something like:
`(Goal a, Goal b) -> a.getName().compareToIgnoreCase(b.getName());`

Reply:
Well this was already like that, so it can stay, the better way would be to use a:
`Comparators.comparing(Goal::getName, String.CASE_INSENSITIVE_ORDER)`